kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/autoconfig/KafkaAutoConfiguration.java

```diff
@@ -91,7 +91,7 @@ public KafkaAutoConfiguration(KafkaProperties properties) {
 
     @Bean("axonKafkaProducerFactory")
     @ConditionalOnMissingBean
-    @ConditionalOnProperty(prefix = "axon.kafka.publisher", name = "enabled", havingValue = "true", matchIfMissing = true)
```

Comment:
I could think of a minor optimization, and that's to *not* create the `kafkaMessageConverter` bean when both publishing and consuming are disabled. Or, we could have a double conditional on this entire auto-config class. Granted, that's a pretty minimal optimization, so not overly pressing. I let you decide whether you want to tackle this, @aupodogov.

Reply:
I did this optimization!